dbt/adapters/bigquery/retry.py

```diff
@@ -1,10 +1,9 @@
 from typing import Callable, Optional
 
-from google.api_core.exceptions import Forbidden
 from google.api_core.future.polling import DEFAULT_POLLING
 from google.api_core.retry import Retry
-from google.cloud.bigquery.retry import DEFAULT_RETRY
-from google.cloud.exceptions import BadGateway, BadRequest, ServerError
+from google.cloud.bigquery.retry import DEFAULT_JOB_RETRY
+from google.cloud.exceptions import BadRequest
```

Comment:
The other exceptions are already captured in the predicate on `DEFAULT_JOB_RETRY`. This is already tested in a unit test (and is how I learned we add this one ourselves).